internal/api/migrate/mapping.cuepart

```diff
@@ -128,52 +128,55 @@ spec: {
         }
     }]
     // go through the top-level panels 1st time, to fill the panels section of Perses
-    panels: { for grafanaPanelId, grafanaPanel in #grafanaDashboard.panels {
-        // function-like pattern to factorize the panel conversion logic
-        _panelConversion: {
-            #panel: _
 
-            kind: "Panel"
-            spec: {
-                display: {
-                    name: [ // switch
-                        if #panel.title != _|_ if #panel.title != "" {
-                            #panel.title
-                        },
-                        "empty"
-                    ][0]
-                    description: *#panel.description | ""
-                }
-                plugin: [ // switch
-                    %(conditional_panels)
-                ][0]
-                // in case of no targets, the resulting empty array will be removed via unmarshaling in Go (omitempty)
-                queries: [ if #panel.targets != _|_ for _, target in #panel.targets {
-                    kind: "TimeSeriesQuery"
-                    spec: {
-                        #target: target
-                        plugin: [ // switch
-                            %(conditional_timeseries_queries)
+    if #grafanaDashboard.panels != null {
```

Comment:
Could you elaborate on the addition of this check? From what I can see when you manually remove the `panels` attribute from a grafana dashboard via the JSON view, after saving it it's appended back by the backend with an empty array as a value, thus I'm not sure it's worth supporting the case where it would be `null`.

Reply:
Ok I see that the case was happening because of our own preprocessing in `rearrangeGrafanaPanelsWithinExpandedRows()` but you already fixed it by initializing the map, thanks for that. So I guess we can remove this check then.

Reply:
Hi @AntoineThebaud thanks for taking a look, the cue schema change is reverted.